### PR TITLE
 #162727484 Implement reading stat feature

### PR DESCRIPTION
**Pull request by abejide001**
#### What does this PR do?
Allows artist to see the number of times his art-work was visited
#### Description of Task to be completed?
- Write test for reading stat functionality
- Implement reading stat controller and route
- Create models and migrations for reading stat
#### How should this be manually tested?
- Run `npm i`
- Start the server by running `npm run start:dev`
- Visit the endpoint `localhost:9000/api/v1/arts/slug` to get the visited number for a particular art
- Run `npm test` to test the app
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[#162727484](https://www.pivotaltracker.com/story/show/162727484)
#### Screenshot
screen shot 2019-01-28 at 11 12 57 am

### CODE REVIEW

File: src/middlewares/ReadingStat.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import models from '../db/models';
+import Response from '../helpers/response';
+
+const { Art } = models;
+
+/** Reading Stat Class */
+class ReadingStatMiddleware {
```
Comment (julietezekwe) [inline src/middlewares/ReadingStat.js]: Please ensure to make the file name and export name and class name same.

**Review by julietezekwe — COMMENTED** [review]
Nice job. But I have some comments and one observation. I see the reading stat created but I did not see the file for retrieving reading start for a single art. or is it coming with the GET ART action?

**Comment by abejide001** [comment]
> Nice job. But I have some comments and one observation. I see the reading stat created but I did not see the file for retrieving reading start for a single art. or is it coming with the GET ART action?

yes it is

File: src/middlewares/ReadingStat.js (hunk 1 above)
Comment (abejide001) [reply src/middlewares/ReadingStat.js]: okay, thanks.

**Review by julietezekwe — APPROVED** [review]

File: src/controllers/ArtsController.js [hunk 2]
```diff
@@ -63,15 +63,17 @@ class ArtsController {
           categoryId,
           featuredImg: mediaFilesArray[0].url
             || process.env.DEFAULT_ARTICLE_IMAGE,
-          status: defaultStatus
+          status: defaultStatus,
+          visited: 0
```
Comment (andela-dbamidele) [inline src/controllers/ArtsController.js]: You can just set a default value for this in your migration/model file.

File: src/db/migrations/20190114155001-create-art.js [hunk 3]
```diff
@@ -1,52 +1,54 @@
...
-    });
-  },
+    },
+    slug: {
+      type: Sequelize.STRING
+    },
+    title: {
+      type: Sequelize.STRING
+    },
+    visited: {
```
Comment (andela-dbamidele) [inline src/db/migrations/20190114155001-create-art.js]: Please remove this from this file and create a separate migration file to handle this.

File: src/middlewares/ReadingStatMiddleware.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+   *
+   * @param {object} req
+   * @param {object} res
+   * @param {object} next
+   * @returns {*} - object containing the message, status and code
+   * @memberof ReadingStatMiddleware
+   */
+  static async getStat(req, res, next) {
+    try {
+      const read = await Art
```
Comment (andela-dbamidele) [inline src/middlewares/ReadingStatMiddleware.js:19]: Add an attribute field to specify the exact field you need from the query.

**Review by andela-dbamidele — CHANGES_REQUESTED** [review]

File: src/middlewares/ReadingStatMiddleware.js (hunk 4 above)
Comment (abejide001) [reply src/middlewares/ReadingStatMiddleware.js:19]: fixed

File: src/db/migrations/20190114155001-create-art.js (hunk 3 above)
Comment (abejide001) [reply src/db/migrations/20190114155001-create-art.js]: fixed

File: src/controllers/ArtsController.js (hunk 2 above)
Comment (abejide001) [reply src/controllers/ArtsController.js]: fixed

**Pull request merged**